packages/grid/x-data-grid/src/models/colDef/gridColDef.ts

```diff
@@ -260,7 +261,7 @@ export interface GridActionsColDef extends GridColDef {
    * @param {GridRowParams} params The params for each row.
    * @returns {React.ReactElement<GridActionsCellItemProps>[]} An array of [[GridActionsCell]] elements.
    */
-  getActions: (params: GridRowParams) => React.ReactElement<GridActionsCellItemProps>[];
+  getActions: (params: GridRowParams<R>) => React.ReactElement<GridActionsCellItemProps>[];
 }
 
 export type GridEnrichedColDef<R extends GridValidRowModel = any, V = any, F = V> =
```

Comment:
You also need to pass the generics to `GridActionsColDef` to make the typings work below:

```ts
const columns: GridColumns<TestRowModel> = [
  {
    field: 'actions',
    type: 'actions',
    getActions: ({ id, row }) => {},
  },
];
```

The diff in https://github.com/mui/mui-x/issues/4660#issuecomment-1109929700 shows how to do.